manifests/init.pp

```diff
@@ -0,0 +1,84 @@
+# Enter documentation here
+# Parameters
+# @param ensure
+#   present (default) will install files and packages
+#   absent  make sure they are not installed.
+# @param manage_client
+#   True = install the net-simp-utils.  These are command line utilities.
+# @param  $autoupgrade
+#   if true packages will be installed with "latest"
+#   if false packages will be installed with "present"
+# @param  $version
+#   The version of snmp protocol to use.
+#   At this time the simp_profile only manages v3, to configure
+#   older versions use the snmp module directly.
+# @param $v3_users_hash,  $view_hash, $group_hash, $access_hash,
+#   These are hashed used to set up access to snmpd.  See
+#   the access.pp module for more information.
+# @param $simp_snmp_file
+#   file to hold snmp configuration directives for client utils.
+# @param $simp_snmpd_dir
+#   Directory to hold configuration files defined by simp and used
+#   by the snmpd daemon.  These files are managed by puppet.
+# @param $user_snmpd_dir
+#   Directory to hold additional configuration files created by user.
+#   These files are not managed by puppet.  For settings that are
+#   one off (and not cumulative like groups or access) the last one wins.
+#   This diretory is read after the simp_snmpd directory and will
+#   override those settings.
+# @param agentaddress
+#   An array of listening addresses for the snmpd to listen on. Each
+#   element of the array is an array of the form
+#   [ Protocol, Address, Port ].
+#
+#   The addresses to have the snmpd listen on.
+
+class simp_snmpd (
+  Enum['present','absent']             $ensure,
+  Integer[3,3]                         $version,
```

Comment:
Instead of requiring 3, let users pass any int and throw a more useful failure message explaining the module only supports 3